pulp_python/tests/functional/api/test_sync.py

```diff
@@ -451,6 +454,228 @@ def test_exclude_unavailable_projects(self):
         )
 
 
+class LatestKeptPackagesTestCase(unittest.TestCase):
+    """
+    Test checks that latest X packages are synced when latest kept is
+    specified
+
+    Targets issue:
+    https://pulp.plan.io/issues/138
+
+    This feature uses Bandersnatch's latest_kept filter which doesn't work well
+    with the other filters like pre-releases and allow/blocklist. Whether to
+    count the behavior as a bug or feature is hard to tell.
+    """
```

Comment:
I might need to add some notes to the docs to describe this filters behavior a bit better.  It's pretty poorly written on the Bandersnatch side and will often only return the latest_version even if latest_kept > 1 because packages tend to have a lot of prerelease versions, but the prerelease filter will remove them.

Reply:
Honestly I think the Bandersnatch filtering could use some re-writing.  It's kind of sub-optimal to grab the latest N z-releases, pre-release or not.  Like, if you want N versions of Django, you would probably rather have 3.2.2, 3.1.9, 3.0.17 than 3.2.2, 3.2.1, 3.2.0

That's a discussion to have upstream though.